scripts/emulator-tests/functionsEmulatorRuntime.spec.ts

```diff
@@ -457,6 +461,42 @@ describe("FunctionsEmulator-Runtime", () => {
 
       expect(res.var).to.eql("localhost:9099");
     }).timeout(TIMEOUT_MED);
+
+    it("should inject user environment variables when preview is enabled", async () => {
+      // Setup
+      const restore = previews.dotenv;
```

Comment:
I'd recommend extracting this out into a describe block with a `beforeEach` and `afterEach` so that if the test fails it still properly restores afterwards.